bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/DiscoveryService.java

```diff
@@ -87,6 +113,20 @@ public interface DiscoveryService {
      */
     void startScan(@Nullable ScanListener listener);
 
+    /**
+     * Triggers this service to start an active scan for new devices using an input parameter for that.<br>
+     * This method must not block any calls such as {@link #abortScan()} and
+     * must return fast.
+     * <p>
+     * If started, any registered {@link DiscoveryListener} must be notified about {@link DiscoveryResult}s.
+     * <p>
+     * If there is already a scan running, it is aborted and a new scan is triggered.
+     *
+     * @param input an input parameter to be used during discovery scan
+     * @param listener a listener that is notified about errors or termination of the scan
+     */
+    void startScanWithInput(String input, @Nullable ScanListener listener);
```

Comment:
Note that I added a new method rather than adding a new parameter to the existing method `startScan` because this method is called by few bindings.
But it is also possible to only update the existing method and prepare a fix for existing bindings.

Reply:
Any reason not to overload the startScan method ? 

```java
void startScan(@Nullable ScanListener listener);
void startScan(String input, @Nullable ScanListener listener);
```

Reply:
Ok, method renamed.

There is still the method `void startScanWithInput(String input)`, the main method that will be overriden in a binding. Before renaming it into startScan, I have to check if we have bindings declaring such method.

Reply:
```

$ find */src -name "*.java" -exec grep -H "void startScan(" {} \; | grep -v "void startScan()"
org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/rio/system/RioSystemHandler.java:    private void startScan(RioSystemConfig sysConfig) {
org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/discovery/WebthingDiscoveryService.java:    private void startScan(boolean isBackground) {
```
Ok, there should be no risk I believe.

Reply:
Every occurrence of `startScanWithInput` is not removed.

Reply:
To come back to my initial comment, in case you would prefer to not add a new method but rather add a parameter to the existing method, I checked how many bindings should be updated. Only five: easee, gardena, homematic, hue and lutron. So not a big deal. Let me know what you prefer.
```
$ find */src -name "*.java" -exec grep -H "startScan(" {} \; | grep -v "startScan()"
org.openhab.binding.easee/src/main/java/org/openhab/binding/easee/internal/handler/EaseeSiteHandler.java:            discoveryService.startScan(null);
org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java:                discoveryService.startScan(null);
org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/handler/HomematicBridgeHandler.java:                discoveryService.startScan(null);
org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java:                discoveryService.startScan(null);
org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/IPBridgeHandler.java:                discoveryService.startScan(null);
org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/discovery/internal/ModbusDiscoveryService.java:            scanStarted |= service.startScan(this);
org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/discovery/internal/ModbusEndpointDiscoveryService.java:    public boolean startScan(ModbusDiscoveryService service) {
org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/discovery/internal/ModbusThingHandlerDiscoveryService.java:    boolean startScan(ModbusDiscoveryService service);
org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/rio/system/RioSystemHandler.java:                    startScan(rioConfig);
org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/rio/system/RioSystemHandler.java:     * {@link #startScan(RioSystemConfig)} when a device should be scanned and 'things' discovered from it
org.openhab.binding.russound/src/main/java/org/openhab/binding/russound/internal/rio/system/RioSystemHandler.java:    private void startScan(RioSystemConfig sysConfig) {
org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/discovery/WebthingDiscoveryService.java:        startScan(true);
org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/discovery/WebthingDiscoveryService.java:    private void startScan(boolean isBackground) {
org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/discovery/WebthingDiscoveryService.java:        startScan(false);
```